### ISSUE TITLE
PTT always on with serial port

### ISSUE THREAD

**Issue by cx5cba** [wb2osz/direwolf#44]
I use serial port to PTT in Windows 7 (Native RS232 and USB to RS232 adapter). When I turn on the Direwolf the TRX turn in TX.
I only made one change in the direwolf.conf:  "PTT COM1 RTS"

I tried the AGWPE and the PTT work fine

Thanks for help

**Comment by dranch** [comment]
Hello CX5CBA,

Assuming the RTS signal is the correct signal, try "PTT COM1 -RTS".  Notice the added "-" which reverses the pin's logic.

--David
KI6ZHD

**Comment by cx5cba** [comment]
Hello David,

I tried that you said but still the error.

Thanks

Carlos
CX5CBA

**Comment by dranch** [comment]
Hmm.. that's very strange.  AGW is configured for the RTS signal on that 
COM port?

--David

**Comment by cx5cba** [comment]
Yes, AGW use RTS for left channel and DTR for right channel. It´s very strange.

**Comment by dranch** [comment]
Oh.. that's an entirely DIFFERENT config.  You'd want something like the 
following but make sure each PTT entry is in the correct CHANNEL section:

CHANNEL 0
PTT COM1 RTS
CHANNEL 1
PTT COM1 DTR

You might have to change RTS / -RTS / DTR / -DTR to match the right 
polarity of your PTT circuit to the radio.

--David
KI6ZHD

**Comment by DwaineGarden** [comment]
I have the same problem.  When using the PTT with USB serial cable the radio is in TX all the time.  I think something has broken because it use to work fine.

**Comment by DwaineGarden** [comment]
It's happening on both raspberrypi and a regular Linux computer.

**Comment by dranch** [comment]
This ticket is over a year old and Direwolf 1.4 has been released with many fixes.  Dwaine, in your environment, which serial line are you using for PTT?  You should be using RTS (don't use DTR).

**Comment by cx5cba** [comment]
"This ticket is over a year old and Direwolf 1.4 has been released with many fixes. Dwaine, in your environment, which serial line are you using for PTT? You should be using RTS (don't use DTR)."

Hello dranch.
I use v1.4 and still the PTT error only in Direwolf.

**Comment by DwaineGarden** [comment]
I'm using -RTS on the dev branch.  I also tried two different USB serial cables on the raspberry Pi.  Same constantly open TX.  It use to work fine with TX on then TX off after the transmit.

**Comment by dranch** [comment]
Please join and post about your issue on direwolf_packet@yahoogroups.com and we'll get your issue solved.  Github's "issues" section is for real bugs/enhancements so let's prove / disprove it there as too many people are successfully using Direwolf w/o issue.

**Comment by cx5cba** [comment]
Dear David,

I successfully using Direwolf with native serial ports and USB adapters but not for FT-991 USB port, Therefore it´s a real issue for me.
Thanks for your cooperation

**Comment by DwaineGarden** [comment]
Ok. I will post to the forum.

**Comment by wb2osz** [comment]
When Windows reboots, it sets serial port RTS to the high state.  
I don't know if there is some way to tell the operating system to make the default low state.
